source/CCustomOpcodeSystem.cpp

```diff
@@ -992,6 +955,116 @@ namespace CLEO
 		return -1; // error
 	}
 
+	OpcodeResult CCustomOpcodeSystem::CleoReturnGeneric(WORD opcode, CRunningScript* thread, bool returnArgs, DWORD returnArgCount, bool strictArgCount)
+	{
+		auto cs = reinterpret_cast<CCustomScript*>(thread);
+
+		ScmFunction* scmFunc = ScmFunction::Get(cs->GetScmFunction());
+		if (scmFunc == nullptr)
+		{
+			SHOW_ERROR("Invalid Cleo Call reference. [%04X] possibly used without preceding [0AB1] in script %s\nScript suspended.", opcode, cs->GetInfoStr().c_str());
+			return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+		}
+
+		// store return arguments
+		static SCRIPT_VAR arguments[32];
+		static bool argumentIsStr[32];
+		std::forward_list<std::string> stringParams; // scope guard for strings
+		if (returnArgs)
+		{
+			if (returnArgCount > 32)
+			{
+				SHOW_ERROR("Opcode [%04X] has too many (%d) args in script %s\nScript suspended.", opcode, returnArgCount, cs->GetInfoStr().c_str());
+				return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+			}
+
+			auto nVarArg = GetVarArgCount(thread);
+			if (returnArgCount > nVarArg)
+			{
+				SHOW_ERROR("Opcode [%04X] declared %d args, but %d was provided in script %s\nScript suspended.", opcode, returnArgCount, nVarArg, ((CCustomScript*)thread)->GetInfoStr().c_str());
+				return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+			}
+
+			std::fill(std::begin(argumentIsStr), std::end(argumentIsStr), 0);
+			for (DWORD i = 0; i < returnArgCount; i++)

```

Comment:
would it be more efficient to assign `argumentIsStr[i] = false` inside the for loop, instead of using three calls for the iterator?

Reply:
```
		for (DWORD i = 0; i < returnArgCount; i++)
		{
			argumentIsStr[i] = false;
			SCRIPT_VAR* arg = arguments + i;
```

Reply:
True. Fixed